Library/Homebrew/language/node.rb

```diff
@@ -0,0 +1,35 @@
+module Language
+  module Node
+    def self.npm_cache_config
+      "cache=#{HOMEBREW_CACHE}/npm_cache\n"
+    end
+
+    def self.setup_npm_environment
+      npmrc = Pathname.new("#{ENV["HOME"]}/.npmrc")
+      # only run setup_npm_environment once per formula
+      return if npmrc.exist?
```

Comment:
Just to check here, is there any chance a project ships an `.npmrc` in the tarball, which we then place in the `buildpath` and consequently this `return` is invoked before the `$PATH` is successfully modified below?